2024 and 2015 NSDUH Dataset Analysis <br>
Xavier McFadden | Data Science for Addiction Research Fellowship | Spring 2026 <br>
Exploratory analysis of stimulant and sedative misuse in the past year between ages 16 and 25 using NSDUH datasets (2024 and 2015 respectively) <br>
Due 3/6/26

In [2]:
#dataset import (2024)
import pandas as pd

df2024 = pd.read_csv("NSDUH_2024_Tab.txt", sep="\t", low_memory=False)


In [ ]:
# first overview of the dataset
print(df2024.shape)

In [ ]:
# filter to relevant columns for analysis
key_cols = ['AGE3', 'STMNMREC', 'SEDNMREC']
df_filtered2024 = df2024[key_cols]

# filter to ages 16-25 (AGE3 values 3-6)
df_filtered2024 = df_filtered2024[df_filtered2024['AGE3'].between(3, 6)]

print(df_filtered2024.head())
print(df_filtered2024.shape)

# past year misuse = codes 1 and 2
print(df_filtered2024['STMNMREC'].value_counts())
print(df_filtered2024['SEDNMREC'].value_counts())

2024 Dataset (ages 16-25): <br>
Stimulant misuse in the past year: <br>
317 + 91 = 408 <br>
Sedative misuse in the past year: <br>
16 + 29 = 45

In [ ]:
# 2015 dataset import and analysis begins here
df2015 = pd.read_csv("NSDUH_2015_Tab.tsv", sep="\t", low_memory=False)

In [ ]:
print(df2015.shape)

In [ ]:
#filter relevant columns for 2015
key_cols_2015 = ['AGE2', 'STMNMREC', 'SEDNMREC']
df_filtered2015 = df2015[key_cols_2015]

#filter to ages 16-25 (AGE2 values 5-12)
df_filtered2015 = df_filtered2015[df_filtered2015['AGE2'].between(5, 12)]

print(df_filtered2015.head())
print(df_filtered2015.shape)


print(df_filtered2015['STMNMREC'].value_counts())
print(df_filtered2015['SEDNMREC'].value_counts())

(19131, 3)
STMNMREC
91    17420
2       873
3       415
1       346
98       36
83       35
9         4
8         2
Name: count, dtype: int64
SEDNMREC
91    18685
3       208
2       104
83       52
1        41
98       39
9         2
Name: count, dtype: int64


2015 Dataset Analysis: <br>
Stimulant misuse in the past year: <br>
346 + 873 = 1219 <br>
Sedative misuse in the past year: <br>
41 + 104 = 145 <br>

2015 -> 2024 <br>
 <br>
Stimulant misuse in the past year: <br>
1219 -> 408 <br>
Sedative misuse in the past year: <br>
145 -> 45 <br>
<br>
Overall, misuse in both categories has decreased by a factor of about 3x. This could be due to a multitude of reasons.